Teapot/Logger.swift

```diff
@@ -0,0 +1,106 @@
+import Foundation
+
+/// Wrapper for logs which can be toggled on and off to increase or decrease the amount of log barf you wish to see.
+public class Logger {
+
+    public enum LogLevel: Int {
+        case
+        // Logs all data sent or received to the console, including the string representation of the data which was sent or received.
+        incomingAndOutgoingData,
+
+        // Logs all data received to the console, including the string representation of the data which was sent.
+        incomingData,
+
+        // Logs any errors to the console, including the string representation of the data which produced them.
+        error,
+
+        // Does not log anything to the console.
+        none
+    }
+
+    /// The current `LogLevel` for this instance of `Logger`.
+    /// NOTE: In production, you should almost certainly use `.none`, which is also the default value.
+    public var currentLevel: LogLevel
+
+    /// Designated initializer.
+    ///
+    /// - Parameter level: The level to log at. Defaults to `.none`
+    public init(level: LogLevel = .none) {
+        self.currentLevel = level
+    }
+
+    /// Logs an item if the log level is `incomingAndOutgoing` or higher.
+    ///
+    /// - Parameters:
+    ///   - items: What you wish to log.
+    ///   - file: The file this log is coming from. Defaults to the directly calling file.
+    ///   - line: The line this log is coming from. Defaults to the directly calling line.
+    ///   - date: The date this
+    /// - Returns: A boolean indicating whether the log printed or not, mostly for testing purposes.
+    @discardableResult
+    public func incomingAndOutgoingDataLog(_ items: @autoclosure () -> String, file: StaticString = #file, line: UInt = #line, at date: Date = Date()) -> Bool {
```

Comment:
fwiw: This and the related method are using `@autoclosure` so that the logs aren't actually generated unless they're going to be printed out. This prevents logs that aren't being printed from killing performance.